incubator/hnc/internal/validators/object.go

```diff
@@ -244,16 +247,17 @@ func (o *Object) handleInherited(op admissionv1beta1.Operation, newSource, oldSo
 	return deny(metav1.StatusReasonInternalError, "unknown operation: "+string(op))
 }
 
-// isDeletingNS returns true if the namespace of the object is already being deleted, based on the
-// in-memory forest.
-func (o *Object) isDeletingNS(inst *unstructured.Unstructured) bool {
-	o.Forest.Lock()
-	defer o.Forest.Unlock()
-	ns := o.Forest.Get(inst.GetNamespace())
-	if ns == nil {
-		return false
+// isDeletingNS returns true if the namespace of the object is already being deleted.
+func (o *Object) isDeletingNS(ctx context.Context, inst *unstructured.Unstructured) bool {
```

Comment:
Now that `isDeletingNS` can fail, I'd prefer it to return an error. That way, if there's a transient error, it will be returned all the way up to the user.